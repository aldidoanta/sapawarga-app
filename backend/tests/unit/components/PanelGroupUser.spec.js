import { shallowMount, mount } from '@vue/test-utils'
import PanelGroup from '@/views/user/components/PanelGroup'
import totalUser from '../fixtures/totalUser'

describe('PanelGroup User', () => {
  let props

  const build = () => {
    const wrapper = mount(PanelGroup, {
      propsData: props
    })
    return {
      wrapper,
      cardColumn: () => wrapper.find('.card-panel-col'),
      cardIcon: () => wrapper.find('.card-panel-icon-wrapper.icon-people-all'),
      cardNumber: () => wrapper.find('.card-panel-description')
    }
  }

  beforeEach(() => {
    props = {
      totalUserSummary: totalUser,
      totalAllUser: totalUser.totalUser,
      duration: 1
    }
  })

  it('renders the component', () => {
    // arrange
    const { wrapper } = build()

    // assert
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders main elements', (done) => {
    // arrange
    const { cardColumn, cardIcon, cardNumber, wrapper } = build()

    // assert
    expect(cardColumn().exists()).toBe(true)
    expect(cardIcon().exists()).toBe(true)
    expect(cardNumber().exists()).toBe(true)

    expect(cardNumber().text()).toContain('0')

    setTimeout(() => {
      expect(wrapper.text()).toContain('Semua')
      expect(wrapper.text()).toContain('99')
      done()
    }, 1100)

    expect(wrapper.props().totalAllUser).toBe(totalUser.totalUser)
    // expect(name().text()).toBe(props.user.name)
  })
})
