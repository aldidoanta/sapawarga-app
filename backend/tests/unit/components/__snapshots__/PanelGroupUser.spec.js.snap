// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelGroup User renders the component 1`] = `
<el-row gutter="40" class="panel-group" totalusersummary="[object Object]">
  <el-card class="box-card">
    <div slot="header" class="clearfix"><span>Total user per:</span></div>
    <div style="margin-bottom: 50px;">
      <el-col xs="12" sm="12" lg="6" class="card-panel-col">
        <div class="card-panel">
          <div class="card-panel-icon-wrapper icon-people-all">
            <svg-icon icon-class="peoples" class-name="card-panel-icon"></svg-icon>
          </div>
          <div class="card-panel-description">
            <div class="card-panel-text">
              Semua
            </div> <span class="card-panel-num">
  0
</span>
          </div>
        </div>
      </el-col>
      <el-col xs="12" sm="12" lg="6" class="card-panel-col">
        <div class="card-panel">
          <div class="card-panel-icon-wrapper icon-people-province">
            <svg-icon icon-class="peoples" class-name="card-panel-icon"></svg-icon>
          </div>
          <div class="card-panel-description">
            <div class="card-panel-text">
              Provinsi
            </div> <span class="card-panel-num">
  0
</span>
          </div>
        </div>
      </el-col>
      <el-col xs="12" sm="12" lg="6" class="card-panel-col">
        <div class="card-panel">
          <div class="card-panel-icon-wrapper icon-people-kota">
            <svg-icon icon-class="peoples" class-name="card-panel-icon"></svg-icon>
          </div>
          <div class="card-panel-description">
            <div class="card-panel-text">
              Kota/Kab
            </div> <span class="card-panel-num">
  0
</span>
          </div>
        </div>
      </el-col>
      <el-col xs="12" sm="12" lg="6" class="card-panel-col">
        <div class="card-panel">
          <div class="card-panel-icon-wrapper icon-people-kec">
            <svg-icon icon-class="peoples" class-name="card-panel-icon"></svg-icon>
          </div>
          <div class="card-panel-description">
            <div class="card-panel-text">
              Kecamatan
            </div> <span class="card-panel-num">
  0
</span>
          </div>
        </div>
      </el-col>
      <el-col xs="12" sm="12" lg="6" class="card-panel-col">
        <div class="card-panel">
          <div class="card-panel-icon-wrapper icon-people-kel">
            <svg-icon icon-class="peoples" class-name="card-panel-icon"></svg-icon>
          </div>
          <div class="card-panel-description">
            <div class="card-panel-text">
              Kelurahan
            </div> <span class="card-panel-num">
  0
</span>
          </div>
        </div>
      </el-col>
      <el-col xs="12" sm="12" lg="6" class="card-panel-col">
        <div class="card-panel">
          <div class="card-panel-icon-wrapper icon-people-rw">
            <svg-icon icon-class="peoples" class-name="card-panel-icon"></svg-icon>
          </div>
          <div class="card-panel-description">
            <div class="card-panel-text">
              RW
            </div> <span class="card-panel-num">
  0
</span>
          </div>
        </div>
      </el-col>
      <el-col xs="12" sm="12" lg="6" class="card-panel-col">
        <div class="card-panel">
          <div class="card-panel-icon-wrapper icon-people-rt">
            <svg-icon icon-class="peoples" class-name="card-panel-icon"></svg-icon>
          </div>
          <div class="card-panel-description">
            <div class="card-panel-text">
              RT
            </div> <span class="card-panel-num">
  0
</span>
          </div>
        </div>
      </el-col>
    </div>
  </el-card>
</el-row>
`;
